src/main/java/decodes/tsdb/TsdbAppTemplate.java

```diff
@@ -105,511 +33,518 @@
+        {
+            Logger.instance().info(appNameArg.getValue() + " exiting.");
+            System.exit(0);
+        }
+    }
+
+    /**
+     * Subclass can override this method if it has any one-time initialization
+     * to do prior to instanting database connections for DECODES and TSDB.
+     */
+    protected void oneTimeInit()
+    {
+        // Empty stub
+    }
+
+    protected void startupLogMessage()
+    {
+        Logger.instance().info("===============================================");
+        Logger.instance().info(appNameArg.getValue() + " starting. "
+            + DecodesVersion.startupTag() + ", pid=" + getPID());
+    }
+
+    /**
+     * Override this and add any program-specific arguments.
+     */
+    protected void addCustomArgs(CmdLineArgs cmdLineArgs)
+    {
+        // Default impl does nothing.
+    }
+
+    /**
+     * Override this with the guts of your program.
+     */
+    protected abstract void runApp()
+        throws Exception;
+
+    /**
+     * Parses the command line arguments.
+     * You probably don't need to override this method. This calls the 
+     * parseArgs method in the CmdLineArgs class. Your app can retrieve the
+     * results later.
+     * @param args the argument from the main method.
+     */
+    protected void parseArgs(String args[])
+        throws Exception
+    {
+        if (!cmdLineArgs.isNoInit())
+            Logger.setLogger(new StderrLogger(appNameArg.getValue()));
+
+        // Parse command line arguments.
+        try { cmdLineArgs.parseArgs(args); }
+        catch(IllegalArgumentException ex)
+        {
+            System.exit(1);
+        }
+        
+        // Remember the application-level debug level set from arguments.
+        appDebugMinPriority = Logger.instance().getMinLogPriority();
+    }
+
+    /**
+     * Creates the TimeSeriesDb object and connects to the underlying SQL
+     * database.
+     * 'theDb' is static in this base-class. Therefore if a sibling in
+     * the launcher has already loaded the database, this method does nothing.
+     * This is also why this method is synchronized.
+     * @throws ClassNotFoundException if can't find database class name
+     * @throws InstantiationException if can't instantiate database object
+     * @throws IllegalAccessException if no permission to access database class
+     */
+    public synchronized void createDatabase()
+        throws ClassNotFoundException,
+        InstantiationException, IllegalAccessException
+    {
+        if (theDb != null)
+            return;
+
+//        String className = DecodesSettings.instance().dbClassName;
+        String className = DecodesSettings.instance().getTsdbClassName();
+
+        if (className == null)
+        {
+            String message = "Error: can't create time series database with editDatabaseType= '"
+                    + DecodesSettings.instance().editDatabaseType + "'";
+            Logger.instance().fatal(message);
```

Comment:
We should change this to slf4j and using the string formatting features.

Reply:
If we construct a message (with concatenation) for purpose of a exception constructor message **and** logging, like the example above  -- Is there still  benefit to string formatting?

Other cases in this file are more clear cut and the slf4j {} formatting is clearly helpful.

Reply:
If I'm doing both I'll usually use String.format. But I've run into this problem myself, should it even do both? Or just throw the exception?

Reply:
Thanks, I'm trying String.format